### PR TITLE
Issue #4642 configure tab size

### PR DESCRIPTION
**Pull request by CruzeBlade**
Hi, 

I added a new preference for the tab size.
Instead of using a tabular char, the user can select to insert spaces instead.
Also, the amount of spaces to be inserted can be configured, in the range of 1 - 8. 
For efficiency reasons, I wouldn't allow inserting any more characters with a single stroke. 

I have currently located this preference under "Preferences > Input System > Tab". 
I'm not very happy about that location, but couldn't find something more intuitive.
In the future, a preference page "Controls" (or something like this) would be good.

Solves #4642.

### CODE REVIEW

**Review by jhilmer — CHANGES_REQUESTED** [review]

File: src/core/gui/dialog/SettingsDialog.cpp
Comment (CruzeBlade) [reply src/core/gui/dialog/SettingsDialog.cpp:389]: I'm not sure, but is the value not already limited by the GtkAdjustment "adjustmentNumberOfSpacesForTab"?
Otherwise, I will add this, of course.

**Comment by jhilmer** [comment]
I would have preferred a separate commit with the clang-format changes.

**Comment by CruzeBlade** [comment]
> I would have preferred a separate commit with the clang-format changes.

Sorry, but the clang-format happened automatically without me noticing it. Should I revert the commit?

**Comment by jhilmer** [comment]
Not needed for me. But it looks like it need to be rebased on master.

**Comment by CruzeBlade** [comment]
Ok, I have rebased. Anything else I need to do?

**Comment by CruzeBlade** [comment]
Hi, would be great if this could be merged. Or do I still need to change / improve something?

**Comment by CruzeBlade** [comment]
Hi @rolandlo, Hi @bhennion 

I would appreciate, if you could take a look at this PR. It appears that it has not received any attention for quite some time.

I completely understand the demands on your time, so please don't feel rushed. I am patient and willing to wait for a response. However, any feedback or guidance you could provide would be immensely helpful.

Thank you for your attention, and I look forward to hearing from you soon.

**Review by rolandlo — CHANGES_REQUESTED** [review]
Thanks for the contribution and sorry for the long wait.
The feature looks good to me and works fine. I would like to suggest two changes to the settings GUI:
- Can you please put the "Number of Spaces: " and spinbox onto the same line like the checkbox? That would look more consistent with the rest of the settings dialog.
- When the "Indent using spaces" checkbox is disabled, the "Number of Spaces" part should be grayed out.

**Comment by CruzeBlade** [comment]
I have changed the UI as suggested:

demo_spaces_as_tab

Does it look like you expected it?

**Comment by rolandlo** [comment]
> I have changed the UI as suggested:
> 
> demo_spaces_as_tab
> 
> Does it look like you expected it?

Yes, that's how I expected it.

**Review by rolandlo — COMMENTED** [review]
It works all fine now. Could you reduce the large diff to the `settings.glade` file though? Only your UI changes should result in changes to that file, not some formatting changes or the use of a different glade version.

Note to self: The content of b5055ab7e48e134997adbfa4024de817dbfeacfb could be simplified using `SettingsDialog::enableWithEnabledCheckbox` from (unmerged) PR #4855.

**Comment by CruzeBlade** [comment]
@rolandlo I would really appreciate it if this could get merged anytime soon.

File: src/core/control/settings/Settings.h
Comment (CruzeBlade) [reply src/core/control/settings/Settings.h]: I have changed it as suggested.

File: src/core/gui/dialog/SettingsDialog.cpp
Comment (CruzeBlade) [reply src/core/gui/dialog/SettingsDialog.cpp]: I have changed it as suggested.

File: src/core/gui/dialog/SettingsDialog.cpp
Comment (CruzeBlade) [reply src/core/gui/dialog/SettingsDialog.cpp:371]: I'm not sure if this cast is still necessary, because ```settings->getNumberOfSpacesForTab()``` is now an unsigned int already.

File: src/core/gui/dialog/SettingsDialog.cpp
Comment (CruzeBlade) [reply src/core/gui/dialog/SettingsDialog.cpp]: I have changed it as suggested.

File: src/core/control/settings/Settings.h
Comment (CruzeBlade) [reply src/core/control/settings/Settings.h]: I have changed it as suggested.

File: src/core/control/settings/Settings.cpp
Comment (CruzeBlade) [reply src/core/control/settings/Settings.cpp]: I hope it's fine now?

File: src/core/control/settings/Settings.h
Comment (CruzeBlade) [reply src/core/control/settings/Settings.h:1165]: Hi, thank you for the review. I appreciate it. I have implemented most of the requested changes. Let me know if everything is okay now.

File: src/core/gui/dialog/SettingsDialog.cpp
Comment (rolandlo) [reply src/core/gui/dialog/SettingsDialog.cpp:371]: No, it's not necessary any more.

**Review by rolandlo — APPROVED** [review]
LGTM

Merging in 24 hours if no objections are raised.

**Comment by rolandlo** [comment]
Ah well, can't merge until @bhennion approves.

**Comment by rolandlo** [comment]
Thanks @CruzeBlade for this useful feature. I just merged it.

**Comment by quantenzitrone** [comment]
hi, was this removed in the gtk4 rewrite?

**Comment by bhennion** [comment]
> hi, was this removed in the gtk4 rewrite?

I don't think it has been removed (or at least not intentionally). Why the question? Does it not work on the GTK4 branch?

**Comment by quantenzitrone** [comment]
> I don't think it has been removed (or at least not intentionally). Why the question? Does it not work on the GTK4 branch?

i am dumb and thought gtk4 was merged into master because i saw some pretty big changes since i last use xournalpp
however i can't find the setting in 1.2.8

**Pull request merged**
